user-service/src/main/java/uk/gov/gchq/palisade/service/user/config/StdUserConfiguration.java

```diff
@@ -49,19 +49,19 @@ public StdUserConfiguration() {
      * @param users a list of objects implementing the {@link UserPrepopulationFactory} interface
      */
     public StdUserConfiguration(final List<StdUserPrepopulationFactory> users) {
-        this.users = users;
+        this.users = Collections.unmodifiableList(users);
     }
 
     @Override
     @Generated
     public List<StdUserPrepopulationFactory> getUsers() {
-        return users;
+        return Collections.unmodifiableList(users);
     }
 
     @Generated
     public void setUsers(final List<StdUserPrepopulationFactory> users) {
         requireNonNull(users);
```

Comment:
Suggest replacing the Object test with the Optional test for a null

Reply:
Updated